### PR TITLE
Kermalis huge refactor #1

### PR DESCRIPTION
**Pull request by Kermalis**
Moved a lot of data out of `data/` and into `src/`. This revealed a lot of file boundaries.

Firstly, every file had "data". These are the `pksdir0` strings everywhere. I think that's obvious, but now it's obvious to me that these are THE FIRST data in the files, and that EVERY file had a data file, empty or not. So I moved a lot of them around so they line up properly now (instead of being at the bottom of some source files).

Also, I started moving things into `src/data/`. Since every file will have a "data" file in the end, we might as well clean it up by putting them in a consistent location. Some files right now have hundreds of lines of const data in the source.c. This is how pokeemerald does it, but imo it will be absolutely awful for this game. I thought about keeping it consistent with pokeemerald, but I really decided against it since this is in fact a different game, so we don't need to do everything like pokeemerald. Since it's cleaner this way, it should be super easy to add translated strings and different-sized windows with `#ifdef`s in these data files. Custom edits of the repo will also be easier if every file has a data file (even if they only have `pksdir0` in them) since it'll be as simple as adding 1 line to the data file and then referencing it in the source file (which was probably chunsoft's workflow). The data in each file is `static` which reveals why there are many duplicated strings in the rom.

Also, the data files aren't the only indication of file boundaries. EWRAM also shows file boundaries directly through the usage of the variables. This ordering is also consistent with the `.text` and `.rodata` but that's not important. I started using the `EWRAM_DATA` and `EWRAM_DATA_2` and `IWRAM_DATA` defines more so we can get more robust file boundaries and be confident about them better. It also reveals some unused variables (possibly from gold/blue rescue team).

Other things I did:
* Cleaned up `0`/`NULL`/`TRUE`/`FALSE` usage
* Removed many `extern` funcs and instead use .h files. It revealed a lot of files that have similar functionality to each other, while also making the repo way less messy
* Used `typedef` for most structs
* Fixed the fakematch in `kecleon_bros.c` that required makefile changes
* Linked decomp.me with my NONMATCHING, as well as for some funcs in .s files that we didn't get to yet
* Marked `NAKED`/`NONMATCHING` funcs that are broken due to `sub_80095E4()` so it's easier to spot them and why they're not matched
* Annotated file boundaries in `sym_ewram.txt`, `sym_ewram2.txt`, and many data files for easy at-a-glance identifying

### CODE REVIEW

**Review by SethBarberee — COMMENTED** [review]
I don't want to remove parameter names from function declarations where we've labeled them like `itemIndex` and etc. If we don't have them labeled or have param_1 or r0, then I'm fine with removing them. LSP/Intellisense uses the header files to give the code suggestions/completions.

For the rest of the stuff, I'm fine with given the discussions on discord.

File: src/data/post_office_guide1.h [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    LEAVE_FOR_RESCUE,
+    SEND_AOK_MAIL,
+    GET_THANK_YOU_MAIL,
+    GETTING_HELP,
+    SEND_SOS_MAIL,
+    RECEIVE_AOK_MAIL,
+    SEND_THANK_YOU_MAIL
+};
+
+//
```
Comment (AnonymousRandomPerson) [inline src/data/post_office_guide1.h:24]: Empty comments.

**Review by AnonymousRandomPerson — CHANGES_REQUESTED** [review]

File: sym_ewram.txt [hunk 2]
```diff
@@ -235,203 +171,204 @@ gUnknown_202D648: /* 202D648 */
...
 	.space 0x4
 	.align 4
 
 gUnknown_202DE30: /* 202DE30 */
 	.space 0x28
 
+/* Seems to be a general u8 buffer for text, mostly used with item names */
 gUnknown_202DE58: /* 202DE58 */
-	.space 0x50
+	.space 80
```
Comment (AnonymousRandomPerson) [inline sym_ewram.txt:258]: Why are a handful of these changed to decimal instead of hex?

**Review by AnonymousRandomPerson — CHANGES_REQUESTED** [review]

File: sym_ewram.txt (hunk 2 above)
Comment (Kermalis) [reply sym_ewram.txt:258]: I changed it only where it's appropriate. Some of those buffers are 80/160/240 bytes long, so showing the 80/160 makes more sense at a glance than 0x50/0xA0 imo

File: include/dungeon_pokemon_sprites.h [hunk 3]
```diff
@@ -3,13 +3,13 @@
 
 #include "position.h"
 
-struct DungeonPokemonStatusSprite
+DungeonPokemonStatusSprite
```
[the remark this answers is not in the document]
Comment (Kermalis) [reply include/dungeon_pokemon_sprites.h:6]: No idea, but fixed now lol

**Comment by Kermalis** [comment]
Should be good now with this latest commit

**Review by AnonymousRandomPerson — APPROVED** [review]

**Review by SethBarberee — CHANGES_REQUESTED** [review]

File: include/friend_area.h [hunk 4]
```diff
@@ -1,41 +1,44 @@
...
-u8 *GetFriendAreaDescription(u8 index);
-u8 GetFriendAreaUnlockCondition(u8 index);
+const u8 *GetFriendAreaDescription(u8 index);
+const u8 *GetFriendAreaName(u8 index);
 s32 GetFriendAreaPrice(u8 index);
-void sub_8092578(u8 *buffer, u8 index, bool8 printPrice);
-const char *GetFriendAreaName(u8 index);
-void sub_809249C(u8 friendArea, u8 clear);
-void sub_8092638(u8 friendArea, struct unkStruct_8092638 *param_2, bool8 checkLeader, bool8 checkDungeon);
+bool8 GetFriendAreaStatus(u8);
```
Comment (SethBarberee) [inline include/friend_area.h:27]:
```suggestion
bool8 GetFriendAreaStatus(u8 index);
```

File: include/friend_area.h [hunk 5]
```diff
@@ -1,41 +1,44 @@
...
+u8 GetFriendAreaUnlockCondition(u8);
```
Comment (SethBarberee) [inline include/friend_area.h:28]:
```suggestion
u8 GetFriendAreaUnlockCondition(u8 index);
```

File: include/friend_area.h [hunk 6]
```diff
@@ -1,41 +1,44 @@
...
+bool8 HasAllFriendAreas(void);
+void InitializeFriendAreas(void);
+void LoadFriendAreas(void);
+u32 ReadSavedFriendAreas(u8 *, s32);
+u32 SaveFriendAreas(u8 *, u32);
+void UnlockFriendArea(u8);
```
Comment (SethBarberee) [inline include/friend_area.h:34]:
```suggestion
void UnlockFriendArea(u8 index);
```

File: include/friend_area.h [hunk 7]
```diff
@@ -1,41 +1,44 @@
...
 
-u32 SaveFriendAreas(u8 *r0, u32 r1);
-u32 ReadSavedFriendAreas(u8 *r0, s32 r1);
+u8 sub_80923D4(s32 target);
+void sub_8092404(u8, u16 *, bool8, bool8);
+void sub_809249C(u8, bool8);
+void sub_8092558(u8 *, u8);
+void sub_8092578(u8 *, u8, bool8);
+void sub_8092638(u8, unkStruct_8092638 *, bool8, bool8);
```
Comment (SethBarberee) [inline include/friend_area.h:41]:
```suggestion
void sub_8092638(u8 friendArea, unkStruct_8092638 *, bool8 checkLeader, bool8 checkDungeon);
```

File: include/moves.h [hunk 8]
```diff
@@ -1,70 +1,92 @@
...
-int unk_FindMarkedMoveInLinkedSequences44(struct Move linkedSequences[4][4]);
-void unk_LinkedSequencesToMoves4(struct Move* moves, struct Move linkedSequences[4][4]);
-int unk_FindMarkedMoveInLinkedSequences88(struct Move linkedSequences[8][8]);
-void unk_LinkedSequencesToMoves8(struct Move* moves, struct Move linkedSequences[8][8]);
-int unk_FindMarkedMoveInLinkedSequences88_v2(struct Move linkedSequences[8][8]);
-void unk_LinkedSequencesToMoves8_v2(struct Move* moves, struct Move linkedSequences[8][8]);
+bool8 CanBeSnatched(u16 moveID);
+void CopyAndResetMoves(Move *, Move *);
+void CopyBareMoveData(Move *, Move *);
+bool8 DoesMoveCharge(u16);
```
Comment (SethBarberee) [inline include/moves.h:27]:
```suggestion
bool8 DoesMoveCharge(u16 moveID);
```

File: include/moves.h [hunk 9]
```diff
@@ -1,70 +1,92 @@
...
+bool8 FailsWhileMuzzled(u16);
```
Comment (SethBarberee) [inline include/moves.h:28]:
```suggestion
bool8 FailsWhileMuzzled(u16 moveID);
```

File: include/moves.h [hunk 10]
```diff
@@ -1,70 +1,92 @@
...
+s32 GetLinkedSequence(s32, Move *, u16 *);
+s32 GetMoveAccuracyOrAIChance(Move *, u32);
+u8 GetMoveAIWeight(Move *);
+s32 GetMoveBasePower(Move *);
+u32 GetMoveBasePP(Move *);
+u32 GetMoveCritChance(Move *);
+u32 GetMoveMaxUpgradeLevel(Move *);
+u32 GetMoveNumberOfChainedHits(Move *);
+s16 GetMoveTargetAndRange(Move *, bool32);
+u8 *GetMoveUseText(u16);
```
Comment (SethBarberee) [inline include/moves.h:40]:
```suggestion
u8 *GetMoveUseText(u16 moveID);
```

File: include/moves.h [hunk 11]
```diff
@@ -1,70 +1,92 @@
...
-void unk_LinkedSequencesToMoves8_v2(struct Move* moves, struct Move linkedSequences[8][8]);
+bool8 CanBeSnatched(u16 moveID);
+void CopyAndResetMoves(Move *, Move *);
+void CopyBareMoveData(Move *, Move *);
+bool8 DoesMoveCharge(u16);
+bool8 FailsWhileMuzzled(u16);
+const u8 *GetHMTMMoves(s16);
+const u8 *GetLevelUpMoves(s16);
+s32 GetLinkedSequence(s32, Move *, u16 *);
+s32 GetMoveAccuracyOrAIChance(Move *, u32);
```
Comment (SethBarberee) [inline include/moves.h:32]:
```suggestion
s32 GetMoveAccuracyOrAIChance(Move *, u32 accuracyType);
```

File: include/moves.h [hunk 12]
```diff
@@ -1,70 +1,92 @@
...
+u8 GetMoveAIWeight(Move *);
+s32 GetMoveBasePower(Move *);
+u32 GetMoveBasePP(Move *);
+u32 GetMoveCritChance(Move *);
+u32 GetMoveMaxUpgradeLevel(Move *);
+u32 GetMoveNumberOfChainedHits(Move *);
+s16 GetMoveTargetAndRange(Move *, bool32);
```
Comment (SethBarberee) [inline include/moves.h:39]:
```suggestion
s16 GetMoveTargetAndRange(Move *, bool32 isAI);
```

File: include/moves.h [hunk 13]
```diff
@@ -1,70 +1,92 @@
...
+u8 *GetMoveUseText(u16);
+u8 GetMoveType(Move *);
+void InitPokemonMove(Move *, u16);
```
Comment (SethBarberee) [inline include/moves.h:42]:
```suggestion
void InitPokemonMove(Move *, u16 moveID);
```

File: include/moves.h [hunk 14]
```diff
@@ -1,70 +1,92 @@
...
+void InitZeroedPPPokemonMove(Move *, u16);
```
Comment (SethBarberee) [inline include/moves.h:43]:
```suggestion
void InitZeroedPPPokemonMove(Move *, u16 moveID);
```

File: include/moves.h [hunk 15]
```diff
@@ -1,70 +1,92 @@
...
+bool8 IsAnyMoveLinked(s32, Move *);
+bool8 IsMoveEnabled(s32, Move *);
```
Comment (SethBarberee) [inline include/moves.h:45]:
```suggestion
bool8 IsMoveEnabled(s32 index, Move *);
```

File: include/moves.h [hunk 16]
```diff
@@ -1,70 +1,92 @@
...
+bool8 IsMoveSet(s32, Move *);
+bool8 IsNextMoveLinked(s32, Move *);
+bool8 IsReflectedByMagicCoat(u16);
```
Comment (SethBarberee) [inline include/moves.h:48]:
```suggestion
bool8 IsReflectedByMagicCoat(u16 moveID);
```

File: include/moves.h [hunk 17]
```diff
@@ -1,70 +1,92 @@
...
+void unk_FixLinkedMovesSetEnabled8(Move *);
+void unk_FixLinkedMovesSetEnabled8_v2(Move *);
+s32 unk_MoveIDPrintMoveDescription(s32, u16, s32, struct subStruct_203B240 **);
+s32 unk_PrintMoveDescription(s32, Move *, s32, struct subStruct_203B240 **);
+s32 unk_SetMoveToFirstInLinkedSequence4(Move *, s32);
+s32 unk_SetMoveToFirstInLinkedSequence8_v2(Move *, s32);
+s32 unk_SetMoveToLastInLinkedSequence4(Move *, s32);
+s32 unk_SetMoveToLastInLinkedSequence8_v2(Move *, s32);
+bool8 UnlinkMovesAfter(s32, Move *);
+void UnSetMove(s32, Move *);
```
Comment (SethBarberee) [inline include/moves.h:76]:
```suggestion
void UnSetMove(s32 index, Move *);
```

File: include/moves.h [hunk 18]
```diff
@@ -1,70 +1,92 @@
...
+void unk_FixLinkedMovesSetEnabled4(Move *);
+void unk_FixLinkedMovesSetEnabled8(Move *);
+void unk_FixLinkedMovesSetEnabled8_v2(Move *);
+s32 unk_MoveIDPrintMoveDescription(s32, u16, s32, struct subStruct_203B240 **);
+s32 unk_PrintMoveDescription(s32, Move *, s32, struct subStruct_203B240 **);
+s32 unk_SetMoveToFirstInLinkedSequence4(Move *, s32);
+s32 unk_SetMoveToFirstInLinkedSequence8_v2(Move *, s32);
+s32 unk_SetMoveToLastInLinkedSequence4(Move *, s32);
+s32 unk_SetMoveToLastInLinkedSequence8_v2(Move *, s32);
+bool8 UnlinkMovesAfter(s32, Move *);
```
Comment (SethBarberee) [inline include/moves.h:75]:
```suggestion
bool8 UnlinkMovesAfter(s32 index, Move *);
```

File: include/moves.h [hunk 19]
```diff
@@ -1,70 +1,92 @@
...
+bool8 IsNextMoveLinked(s32, Move *);
+bool8 IsReflectedByMagicCoat(u16);
+bool8 IsSoundMove(Move *);
+void LoadWazaParameters(void);
+bool8 MoveCannotHitFrozen(Move *);
+bool8 MoveIgnoresTaunted(Move *);
+void RemoveLinkSequenceFromMoves8(Move *, s32);
+void RestorePokemonMoves(struct unkStruct_8094924 *, Move *);
+void SavePokemonMoves(struct unkStruct_8094924 *, Move *);
+bool8 ToggleMoveEnabled(s32, Move *);
```
Comment (SethBarberee) [inline include/moves.h:56]:
```suggestion
bool8 ToggleMoveEnabled(s32 index, Move *);
```

File: include/moves.h [hunk 20]
```diff
@@ -1,70 +1,92 @@
...
+bool8 ToggleSetMove(s32, Move *);
```
Comment (SethBarberee) [inline include/moves.h:57]:
```suggestion
bool8 ToggleSetMove(s32 index, Move *);
```

File: include/moves.h [hunk 21]
```diff
@@ -1,70 +1,92 @@
...
+bool8 TryLinkMovesAfter(s32, Move *);
```
Comment (SethBarberee) [inline include/moves.h:58]:
```suggestion
bool8 TryLinkMovesAfter(s32 index, Move *);
```

File: include/moves.h [hunk 22]
```diff
@@ -1,70 +1,92 @@
...
+u32 GetMoveNumberOfChainedHits(Move *);
+s16 GetMoveTargetAndRange(Move *, bool32);
+u8 *GetMoveUseText(u16);
+u8 GetMoveType(Move *);
+void InitPokemonMove(Move *, u16);
+void InitZeroedPPPokemonMove(Move *, u16);
+bool8 IsAnyMoveLinked(s32, Move *);
+bool8 IsMoveEnabled(s32, Move *);
+bool8 IsMoveSet(s32, Move *);
+bool8 IsNextMoveLinked(s32, Move *);
```
Comment (SethBarberee) [inline include/moves.h:47]:
```suggestion
bool8 IsNextMoveLinked(s32 index, Move *);
```

File: include/moves.h [hunk 23]
```diff
@@ -1,70 +1,92 @@
...
-void unk_LinkedSequencesToMoves8(struct Move* moves, struct Move linkedSequences[8][8]);
-int unk_FindMarkedMoveInLinkedSequences88_v2(struct Move linkedSequences[8][8]);
-void unk_LinkedSequencesToMoves8_v2(struct Move* moves, struct Move linkedSequences[8][8]);
+bool8 CanBeSnatched(u16 moveID);
+void CopyAndResetMoves(Move *, Move *);
+void CopyBareMoveData(Move *, Move *);
+bool8 DoesMoveCharge(u16);
+bool8 FailsWhileMuzzled(u16);
+const u8 *GetHMTMMoves(s16);
+const u8 *GetLevelUpMoves(s16);
```
Comment (SethBarberee) [inline include/moves.h:30]:
```suggestion
const u8 *GetLevelUpMoves(s16 species);
```

File: include/moves.h [hunk 24]
```diff
@@ -1,70 +1,92 @@
...
-int unk_FindMarkedMoveInLinkedSequences88(struct Move linkedSequences[8][8]);
-void unk_LinkedSequencesToMoves8(struct Move* moves, struct Move linkedSequences[8][8]);
-int unk_FindMarkedMoveInLinkedSequences88_v2(struct Move linkedSequences[8][8]);
-void unk_LinkedSequencesToMoves8_v2(struct Move* moves, struct Move linkedSequences[8][8]);
+bool8 CanBeSnatched(u16 moveID);
+void CopyAndResetMoves(Move *, Move *);
+void CopyBareMoveData(Move *, Move *);
+bool8 DoesMoveCharge(u16);
+bool8 FailsWhileMuzzled(u16);
+const u8 *GetHMTMMoves(s16);
```
Comment (SethBarberee) [inline include/moves.h:29]:
```suggestion
const u8 *GetHMTMMoves(s16 species);
```

**Review by SethBarberee — CHANGES_REQUESTED** [review]
Comment from before still applies on keeping parameter names on functions when they are already labeled. I added some suggestions but take this advice to do it on all of the header.

**Review by SethBarberee — APPROVED** [review]

**Review by SethBarberee — APPROVED** [review]

**Pull request merged**